bot/bot.py

```diff
@@ -33,6 +33,16 @@ def __init__(self, *args, **kwargs):
         self._resolver = None
         self._guild_available = asyncio.Event()
 
+        statsd_url = constants.Bot.statsd_host
+
+        if DEBUG_MODE:
+            # Since statsd is UDP, there are no errors for sending to a down port.
+            # For this reason, setting the statsd host to 127.0.0.1 for development
+            # will effectively disable stats.
+            statsd_url = "127.0.0.1"
+
+        self.stats = statsd.StatsClient(statsd_url, 8125, prefix="bot")
```

Comment:
How does this communicate? Is it async? Should we care?

Reply:
That's something I've been looking into. It's not async but the method for sending stats is basically a UDP payload like `statname:1|c` to increment a counter. As far as I can tell there is little way for this to block.

https://github.com/jsocol/pystatsd/blob/1c90b9fdf322680e2625da659abc2aa5d79b5bff/statsd/client/udp.py#L14-L24

Reply:
Did some digging:

https://github.com/jsocol/pystatsd/issues/93
https://pypi.org/project/aiomeasures/ is async but not being updated for 5 years is concerning.

Reply:
I found the same thing and struggled to make it work as well as `statsd`. If this is a huge issue then statsd is such a simple protocol I could probably write a custom implementation.

Reply:
OK, I am fine with trying it as is. But how could we measure if being synchronous is an issue?

Reply:
That's hard to do, I don't foresee any issues because there is no waiting on a response from statsd and I can't think of any slowdowns over the local Docker network.

Reply:
Async stats client has been implemented in https://github.com/python-discord/bot/commit/bca47a1c2e59fb112b947876cea1836879ac7282 so this should not be a concern.